app-backend/db/src/main/scala/TaskDao.scala

```diff
@@ -591,4 +594,48 @@ object TaskDao extends Dao[Task] {
            WHERE annotation_project_id = ${fromProject}
       """).update.run
   }
+
+  def getMostRecentStatus(task: Task): ConnectionIO[TaskStatus] =
+    getTaskActions(task.id) map { actions =>
+      actions.toNel map { actions =>
+        actions.tail.foldLeft(actions.head)(
+          (action1: TaskActionStamp, action2: TaskActionStamp) => {
+            if (action1.timestamp.isAfter(action2.timestamp)) action1
+            else action2
+          }
+        )
+      } map { _.fromStatus } getOrElse { TaskStatus.Unlabeled }
```

Comment:
We don't want the current status -- the current status is going to be a `*_IN_PROGRESS`, since this task is stuck. We want the second most recent status, which is whatever it transitioned from to get where it is right now.